src/Runtime/LoopIterator.php

```diff
@@ -34,7 +34,18 @@ final class LoopIterator implements \Iterator
 
     public function __construct($seq)
     {
-        $this->seq = is_iterable($seq) ? (is_array($seq) ? new \ArrayIterator($seq) : $seq) : new \ArrayIterator([]);
+        if (is_array($seq)) {
+            $this->seq = new \ArrayIterator($seq);
+        } elseif ($seq instanceof \IteratorAggregate) {
+            do {
+                $seq = $seq->getIterator();
+            } while ($seq instanceof \IteratorAggregate);
+            $this->seq = $seq;
+        } elseif (is_iterable($seq)) {
+            $this->seq = $seq;
```

Comment:
Here $seq is more than probably `Iterator` or `IteratorAggregate` **but** that's almost "internal"...

> Internal (built-in) classes that implement this interface can be used in a [foreach](https://www.php.net/manual/en/control-structures.foreach.php) construct and **do not need to implement** [IteratorAggregate](https://www.php.net/manual/en/class.iteratoraggregate.php) or [Iterator](https://www.php.net/manual/en/class.iterator.php).

Not sure if this could happen IRL .. but maybe a `elseif ($seq instanceof \Iterator) ` would be more precise here ?